src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/tke_top_boundary_condition.jl

```diff
@@ -0,0 +1,91 @@
+using Oceananigans.BoundaryConditions: DiscreteBoundaryFunction, BoundaryCondition, Flux
+
+struct TKETopBoundaryConditionParameters{C, U}
+    top_tracer_boundary_conditions :: C
+    top_velocity_boundary_conditions :: U
+end
+
+const TKEBoundaryFunction = DiscreteBoundaryFunction{<:TKETopBoundaryConditionParameters}
+const TKEBoundaryCondition = BoundaryCondition{<:Flux, <:TKEBoundaryFunction}
+
+@inline Adapt.adapt_structure(to, p::TKETopBoundaryConditionParameters) =
+    TKETopBoundaryConditionParameters(adapt(to, p.top_tracer_boundary_conditions),
+                                      adapt(to, p.top_velocity_boundary_conditions))
+
+@inline on_architecture(to, p::TKETopBoundaryConditionParameters) =
+    TKETopBoundaryConditionParameters(on_architecture(to, p.top_tracer_boundary_conditions),
+                                      on_architecture(to, p.top_velocity_boundary_conditions))
+
+@inline getbc(bc::TKEBoundaryCondition, i::Integer, j::Integer, grid::AbstractGrid, clock, fields, clo, buoyancy) =
+    bc.condition.func(i, j, grid, clock, fields, bc.condition.parameters, clo, buoyancy)
+
+@inline getbc(bc::TKEBoundaryCondition, i::Integer, j::Integer, k::Integer, grid::AbstractGrid, clock, fields, clo, buoyancy) =
+    bc.condition.func(i, j, k, grid, clock, fields, bc.condition.parameters, clo, buoyancy)
+
+"""
+    top_tke_flux(i, j, grid, clock, fields, parameters, closure, buoyancy)
+
+Compute the flux of TKE through the surface / top boundary.
+Designed to be used with TKETopBoundaryConditionParameters in a FluxBoundaryCondition, eg:
+
+```
+top_tracer_bcs = top_tracer_boundary_conditions(grid, tracer_names, user_bcs)
+top_velocity_bcs = top_velocity_boundary_conditions(grid, user_bcs)
+parameters = TKETopBoundaryConditionParameters(top_tracer_bcs, top_velocity_bcs)
+top_tke_bc = FluxBoundaryCondition(top_tke_flux, discrete_form=true, parameters=parameters)
+```
+
+See the implementation in catke_equation.jl.
```

Comment:
> See the implementation in catke_equation.jl.

Is this file still the right place to look for implementation?

Reply:
yes